Format endnote and endexample markers

BillWagner opened the pull request:
The *end note* and *end example* marker for a note should be at the end of a single paragraph note. If a note spans multiple paragraphs (including notes that contain examples or lists), the *end note* or *end example* marker should be its own paragraph.

I searched for both *end note* (first commit) and *end example* (second commit) and made the updates to those that didn't follow the format we agreed on.

BillWagner committed:
format *end note*

The *end note* marker for a note should be at the end of a single paragraph note. If a note spans multiple paragraphs (including notes that contain examples or lists), the *end note* marker should be its own paragraph.

BillWagner committed:
Format *end example*

The *end example* marker for a note should be at the end of a single paragraph note. If a note spans multiple paragraphs (including notes that contain examples or lists), the *end example* marker should be its own paragraph.

BillWagner committed:
fix lint warnings.

jskeet reviewed (commented):
Any idea which file is causing the Word converter to fail? I'm happy to pull your PR and run it locally if that would help.

BillWagner commented:
@jskeet  I'm looking into it this morning. I've already got this PR local, so I'll track it down.

BillWagner committed:
fix markdown converter issues

One example and one note had their final paragraph not included in the block quote.

BillWagner commented:
Found it! (actually two issues).

The good news is the word converter worked correctly, and helped find two additional formatting issues.

jskeet commented on tools/MarkdownConverter/Spec/MarkdownSpec.cs:
```diff
@@ -247,6 +247,9 @@ private static string RemoveBlockComments(string text, string file)
         /// only works at the top level at the moment.
         /// </summary>
+        /// <remarks>
+        /// This may no longer be necessary now that we're using Markdown Lint to avoid this condition.
```
I think it probably *is* still necessary, when they're in separate paragraphs. For example:

```markdown
> *Example*:
>
> Some text in multiple paragraphs.
>
> *end example*

> *Note*:
>
> Some text in multiple paragraphs.
>
> *end note*
```

I *think* the two will be collapsed so we end up with an "end example Note:" paragraph without this. I can add a test for it once this PR is in, perhaps.

BillWagner replied:
I added a comment in the code pointing to this discussion for reference.

jskeet reviewed (approved):
Excellent... I think we're nearly ready to give Rex a document...

BillWagner committed:
add link to comment on code change

This code change may be wrong. See comment.
